File: Ruby/P4/Irrgarten/labyrinth_character.rb

#encoding:utf-8

=begin

    Autor: Isabel Morro Tabares 

    Funcionalidades de los personajes de juego
    De esta heredan Monster y Player

=end

module Irrgarten
    class LabyrinthCharacter

        def initialize (name, intelligence, strength, health)
            @name = name
            @intelligence = intelligence
            @strength = strength
            @health = health

            @row = -1
            @col = -1
        end

        def copia(other)
            set_pos(other.row, other.col)
        end

        def dead
            (@health <= 0)
        end

        def row
            @row
        end

        def col
            @col
        end

        def intelligence
            @intelligence
        end

        def strength
            @strength
        end

        def health
            @health
        end

        def name 
            @name
        end

        def set_health (health)
            @health = health
        end
        
        def set_pos (row, col)
            @row = row
            @col = col
        end

        def to_s
            cadena = "[ Nombre: " + @name + ","
            cadena  += " Inteligencia: " + @intelligence.to_s + ","
            cadena  += " Fuerza: " + @strength.to_s + ","
            cadena  += " Salud: " + @health.to_s + ","
            cadena  += " Posicion: (" + @row.to_s + ", " + @col.to_s + ") "
            cadena
        end

        def got_wounded
            if (@health > 0)
                @health -= 1
            end
        end

    end
end

File: Java/P5/P5/irrgarten/irrgarten/src/irrgarten/LabyrinthCharacter.java

package irrgarten;

/**
 * @brief  Representa a los diferentes tipos de jugadores del juego
 * @author Isabel  Morro Tabares
 */

abstract class LabyrinthCharacter {
    
    private String name;
    private float intelligence;
    private float strength;
    private float health;
    private int row;
    private int col;
    
    public LabyrinthCharacter (String name, float intelligence, float strength, float health){
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        this.health = health;
    }
    
    public LabyrinthCharacter(LabyrinthCharacter other){
        this (other.name, other.intelligence, other.strength, other.health);
        setPos(other.row, other.col);
    }
    
    public boolean dead(){
        return health <= 0;
    }
    
    public int getRow(){
        return row;
    }
    
    public int getCol(){
        return col;
    }
    
    protected float getIntelligence(){
        return intelligence;
    }
    
    protected float getStrength(){
        return strength;
    }
    
    protected float getHealth(){
        return health;
    }
    
    protected void setHealth(float health){
        this.health = health;
    }
    
    public void setPos (int row, int col){
        this.row = row;
        this.col = col;
    }
    
    public void setName(String name){
        this.name = name;
    }
    
    public String getName(){
        return name;
    }
    
    @Override
    public String toString(){
        String cadena;
        cadena = "[ Nombre: " + name + ","
                 + " Inteligencia: " + intelligence + ","
                 + " Fuerza: " + strength + ","
                 + " Salud: " + health + ","
                 + " Posicion: (" + row + ", " + col + ")";
        return cadena;
    }
    
    protected void gotWounded(){
        if (health > 0)
            health--;
    }
    
    public abstract float attack();
    public abstract boolean defend (float attack);
    
}
